BitwardenShared/UI/Vault/VaultItem/ViewItem/ViewItemProcessor.swift

```diff
@@ -199,10 +218,45 @@ private extension ViewItemProcessor {
     ///   - field: The field being copied.
     ///
     private func copyValue(_ value: String, _ field: CopyableField?) {
+        guard case let .data(cipherState) = state.loadingState else {
+            services.errorReporter.log(
+                error: ActionError.dataNotLoaded("Cannot copy value for non-loaded item.")
+            )
+            return
+        }
+
         services.pasteboardService.copy(value)
 
         let localizedFieldName = field?.localizedName ?? Localizations.value
         state.toast = Toast(text: Localizations.valueHasBeenCopied(localizedFieldName))
+        switch field {
+        case .customHiddenField:
+            Task {
+                await services.eventService.collect(
+                    eventType: .cipherClientCopiedHiddenField,
+                    cipherId: cipherState.cipher.id,
+                    errorReporter: services.errorReporter
+                )
+            }
+        case .password:
+            Task {
+                await services.eventService.collect(
+                    eventType: .cipherClientCopiedPassword,
+                    cipherId: cipherState.cipher.id,
+                    errorReporter: services.errorReporter
+                )
+            }
+        case .securityCode:
+            Task {
+                await services.eventService.collect(
+                    eventType: .cipherClientCopiedCardCode,
+                    cipherId: cipherState.cipher.id,
+                    errorReporter: services.errorReporter
+                )
+            }
+        default:
+            break
+        }
```

Comment:
🎨 Given that these calls are the same outside of the event type, maybe we could add a method that logs an event for the cipher in this processor? We could also add an extension to `CopyableField` that returns the event associated with copying that field?